pkg/auto/wordpress/job/meter.go

```diff
@@ -32,43 +32,32 @@ func getAllRecords(ctx context.Context, historyFn listMeterReadingFn, meter stri
 		})
 
 		if err != nil {
-			return nil, err
+			return [2]*gen.MeterReadingRecord{}, err
 		}
 
-		records = append(records, resp.GetMeterReadingRecords()...)
+		for _, record := range resp.GetMeterReadingRecords() {
+			if record.GetRecordTime().AsTime().Before(earliest.GetRecordTime().AsTime()) {
+				earliest = record
+			}
+
+			if record.GetRecordTime().AsTime().After(latest.GetRecordTime().AsTime()) {
+				latest = record
+			}
+		}
 
 		if resp.GetNextPageToken() == "" {
 			break
 		}
 
 		pageToken = resp.GetNextPageToken()
-
 	}
 
-	return records, nil
+	return [2]*gen.MeterReadingRecord{earliest, latest}, nil
 }
 
-func processMeterRecords(multiplier float32, records []*gen.MeterReadingRecord) float32 {
-	if len(records) < 2 {
-		return 0
-	}
-
-	slices.SortFunc(records, func(i, j *gen.MeterReadingRecord) int {
-		timeI := i.GetRecordTime().AsTime()
-		timeJ := j.GetRecordTime().AsTime()
-
-		if timeI.Before(timeJ) {
-			return 1
-		}
-		if timeI.After(timeJ) {
-			return -1
-		}
-		return 0 // If times are equal
-	})
-
-	latest := records[0]
-	earliest := records[len(records)-1]
+func processMeterRecords(multiplier float32, records [2]*gen.MeterReadingRecord) float32 {
```

Comment:
Similarly this should accept `earliest, latest *gen.MeterReadingRecord` instead of an array